frappe/public/js/frappe/ui/filters/field_select.js

```diff
@@ -115,7 +115,11 @@ frappe.ui.FieldSelect = class FieldSelect {
 		}
 
 		// main table
-		var main_table_fields = std_filters.concat(frappe.meta.docfield_list[me.doctype]);
+		var main_table_fields = this.only_link_fields
+			? std_filters
+					.concat(frappe.meta.docfield_list[me.doctype])
+					.filter((df) => df.fieldtype === "Link")
+			: std_filters.concat(frappe.meta.docfield_list[me.doctype]);
```

Comment:
DRY:

var main_table_fields = std_filters.concat(frappe.meta.docfield_list[me.doctype]);
if (this.only_link_field) {
	main_table_fields = main_table_fields.filter((df) => df.fieldtype === "Link");
}